apis/filesystem/src/main/java/org/jclouds/filesystem/predicates/validators/internal/FilesystemBlobKeyValidatorImpl.java

```diff
@@ -38,6 +38,8 @@ public void validate(String name) throws IllegalArgumentException {
         //blobkey cannot start with / (or \ in Windows) character
         if (name.startsWith("\\") || name.startsWith("/"))
             throw new IllegalArgumentException("Blob key '" + name + "' cannot start with \\ or /");
+        if (name.contains("../"))
+            throw new IllegalArgumentException("Blob key '" + name + "' cannot contain ../");
```

Comment:
Why is this different than the container name validator?  For robustness should this tokenize the path via `/` then check each component to see if one contains `.` or `..`?  This would allow keys like `..foo` to work.

Reply:
I think `..foo` should be allowed. I tried several object storage services, such as gcloud and aliyun oss, they all allow `..foo` as object key.